### ISSUE TITLE
[BUG] Stuck signing in

### ISSUE THREAD

**Issue by bamurtaugh** [microsoft/vscode-dotnettools#124]
### Describe the Issue

I try to sign in to the C# Dev Kit extension, and I seem to get stuck part way.

### Steps To Reproduce

1. Open extension's walkthrough, select `Connect` from first step to log into my Microsoft account
2. Sign in in the browser
3. Browser is stuck loading at the `https://login.microsoftonline.com/...` URL, and the VS Code Accounts menu still indicates I need to sign in 
![image](https://github.com/microsoft/vscode-dotnettools/assets/25310137/90157117-765d-4849-90b3-8e9abdad81a4)
4. If I select `Connect` again, it almost seems I'm connected, though the Accounts menu still indicates I need to sign in
![image](https://github.com/microsoft/vscode-dotnettools/assets/25310137/b6c2fe13-24f0-485b-8a8b-7d7633b181e3)

### Expected Behavior

Successful sign in.

### Environment Information

VS Code Version:

Version: 1.80.0-insider (user setup)
Commit: ef6abe4cd1573689d070b8715e97019cfadaeb51
Date: 2023-06-12T05:21:26.842Z
Electron: 22.5.5
Chromium: 108.0.5359.215
Node.js: 16.17.1
V8: 10.8.168.25-electron.0
OS: Windows_NT x64 10.0.25387

OS: Windows 11 Insiders

**Comment by bamurtaugh** [comment]
cc @isidorn for awareness

**Comment by timheuer** [comment]
So the toast goes away, but the profile menu is still showing a bad state? If you reload window does the profile still show needing to sign in? And by your screenshot it looks like you signed in via the walkthrough -- correct? /cc @AndreyTretyak

**Comment by bamurtaugh** [comment]
> So the toast goes away, but the profile menu is still showing a bad state? 

Yes.

> If you reload window does the profile still show needing to sign in? 

I just updated and reloaded VS Code with the latest Insiders, and the profile no longer shows I need to sign in! And if I click into the accounts menu, it shows the C# Dev Kit is now a trusted signed in extension.

> And by your screenshot it looks like you signed in via the walkthrough -- correct?

Yes.

***

This issue is now resolved for me after updating and reloading VS Code, but I'll keep it open in case it's helpful to investigate why it required a reload for the account state to update.
